src/components/List.js

```diff
@@ -1,39 +1,42 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { FirestoreDocument } from 'react-firestore';
-import ListItem from './ListItem';
-import '../css/components/ItemsList.css';
+import SortedList from './SortedList';
 import { getToken } from '../lib/TokenService';
 import AddButton from './AddButton';
+import { filter as removePunctuation } from '../lib/helper';
+import '../css/components/ItemsList.css';
 
 export default function List() {
+  const [searchTerm, setSearchTerm] = useState('');
+
   return (
     <div>
-      <h1>LIST OF ITEMS</h1>
+      <h3>LIST OF ITEMS</h3>
       <FirestoreDocument
         path={`shoppingLists/${getToken()}`}
         render={({ isLoading, data }) => {
-          const itemsKeys = data
-            ? Object.keys(data).filter((key) => key !== 'id')
-            : '';
+          let itemsKeys = data ? getItemsKeys(data, searchTerm) : [];
           return isLoading ? (
             <div className="m-auto">Loading</div>
           ) : (
             <div>
-              {itemsKeys.length >= 1 ? (
-                <ul>
-                  {itemsKeys
-                    .sort((a, b) => (a > b ? 1 : -1))
-                    .map((key) => {
-                      return (
-                        <ListItem
-                          listItem={data[key]}
-                          key={key}
-                          listId={data.id}
-                          itemId={key}
-                        />
-                      );
-                    })}
-                </ul>
+              {itemsKeys.length || searchTerm ? (
+                <>
+                  <div className="search-term-wrapper">
+                    <input
+                      value={searchTerm}
+                      onChange={(e) => setSearchTerm(e.target.value.trim())}
```

Comment:
I think no need to call `trim()` here because it's already done in the `filter `function (`removePunctuation`)

Reply:
We actually had that problem with the text input and was fixed when we added `trim()`. I can't remember it now tbh but I'll try to look at the code again tomorrow to remember why we added `trim()` here. 😅